storage/range.go

```diff
@@ -375,9 +375,13 @@ func (r *Range) redirectOnOrAcquireLeaderLease(timestamp proto.Timestamp) error
 		return r.newNotLeaderError()
 	} else if !held || expired {
 		// Otherwise, if not held by this replica or expired, request renewal.
-		if err := r.requestLeaderLease(timestamp); err != nil {
-			return err
+		err := r.requestLeaderLease(timestamp)
+		// Getting a LeaseRejectedError back means someone else got there
+		// first. Since it's committed by now, we can return a NotLeaderError.
+		if _, ok := err.(*proto.LeaseRejectedError); ok {
```

Comment:
I probably wasn't clear enough above. `err` is almost always nil, and only if it isn't AND it's pointing at a `*proto.LeaseRejectedError` we are entering the `if`.